src/ui/windows/TogglDesktop/TogglDesktop/ui/ViewModels/TimelineBlockViewModel.cs

```diff
@@ -72,6 +77,11 @@ public TimeEntryBlock(Toggl.TogglTimeEntryView te, int hourHeight, DateTime date
             DateCreated = date;
             _timeEntry = te;
             OpenEditView = ReactiveCommand.Create(() => Toggl.Edit(TimeEntryId, false, Toggl.Description));
+            var isNotRunningObservable = Observable.Return(_timeEntry.DurationInSeconds >= 0);
+            ContinueEntry = ReactiveCommand.Create(() => Toggl.Continue(TimeEntryId), isNotRunningObservable);
+            CreateFromEnd = ReactiveCommand.Create(() => TimelineUtils.CreateAndEditTimeEntry(Ended, Ended + 3600), isNotRunningObservable);
+            StartFromEnd = ReactiveCommand.Create(() => TimelineUtils.CreateAndEditRunningTimeEntryFrom(Ended), isNotRunningObservable);
+            Delete = ReactiveCommand.Create(() => Toggl.DeleteTimeEntry(TimeEntryId));

```

Comment:
I can do it, of course, though I like more the current solution, because it doesn't require to handle the case, when `ActiveTimeEntryBlock` is null.

Reply:
I give it a pass because the potential perf benefit seems relatively small here, and because we don't care about 100+ TEs a day cases.
I don't necessarily agree with the principle - null handling is quite trivial. And currently, such potential null errors are just swallowed by XAML - it's not very obvious edge cases handling either.